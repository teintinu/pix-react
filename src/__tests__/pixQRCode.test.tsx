import React from 'react'
import renderer, { act } from 'react-test-renderer'
import { PixQRCode } from '..'

describe('Pix QRCode', () => {
  it('QRCode with default arguments', () => {
    const component = renderer.create(
      <PixQRCode
        pixParams={{
          chave: 'recebedor@email.com',
          recebedor: 'Nome do reebedor',
          cidade: 'Nome da Cidade',
          identificador: '123',
          valor: 123.45,
          mensagem: 'minha mensagem'
        }}
        includeMargin={true}
        size={256}
      />
    )

    expect(component.toJSON()).toMatchSnapshot()
    act(() => {})
    expect(component.toJSON()).toMatchSnapshot()
  })

  it('Error in pixParams', () => {
    const component = renderer.create(
      <PixQRCode
        pixParams={{
          chave: '',
          recebedor: 'Nome do reebedor',
          cidade: 'Nome da Cidade',
          identificador: '123',
          valor: 123.45,
          mensagem: 'minha mensagem'
        }}
        includeMargin={true}
        size={256}
      />
    )

    expect(component.toJSON()).toMatchSnapshot()
    act(() => {})
    expect(component.toJSON()).toMatchSnapshot()
  })
})
