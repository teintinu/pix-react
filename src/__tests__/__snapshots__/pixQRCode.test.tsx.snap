// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pix QRCode Error in pixParams 1`] = `
<span>
  Chave PIX é obrigatória
</span>
`;

exports[`Pix QRCode Error in pixParams 2`] = `
<span>
  Chave PIX é obrigatória
</span>
`;

exports[`Pix QRCode QRCode with default arguments 1`] = `
<svg
  height={256}
  viewBox="0 0 29 29"
  width={256}
>
  <path
    d="M0,0 h29v29H0z"
    fill="#FFFFFF"
    shapeRendering="crispEdges"
  />
  <path
    d="M4 4h7v1H4zM13 4h1v1H13zM18,4 h7v1H18zM4 5h1v1H4zM10 5h1v1H10zM14 5h2v1H14zM18 5h1v1H18zM24,5 h1v1H24zM4 6h1v1H4zM6 6h3v1H6zM10 6h1v1H10zM12 6h5v1H12zM18 6h1v1H18zM20 6h3v1H20zM24,6 h1v1H24zM4 7h1v1H4zM6 7h3v1H6zM10 7h1v1H10zM12 7h3v1H12zM16 7h1v1H16zM18 7h1v1H18zM20 7h3v1H20zM24,7 h1v1H24zM4 8h1v1H4zM6 8h3v1H6zM10 8h1v1H10zM13 8h3v1H13zM18 8h1v1H18zM20 8h3v1H20zM24,8 h1v1H24zM4 9h1v1H4zM10 9h1v1H10zM13 9h1v1H13zM15 9h2v1H15zM18 9h1v1H18zM24,9 h1v1H24zM4 10h7v1H4zM12 10h1v1H12zM14 10h1v1H14zM16 10h1v1H16zM18,10 h7v1H18zM7 12h2v1H7zM10 12h2v1H10zM16 12h1v1H16zM21 12h2v1H21zM4 13h2v1H4zM9 13h1v1H9zM11 13h2v1H11zM15 13h2v1H15zM19 13h3v1H19zM23,13 h2v1H23zM4 14h2v1H4zM9 14h4v1H9zM16 14h4v1H16zM21 14h1v1H21zM24,14 h1v1H24zM5 15h1v1H5zM7 15h2v1H7zM12 15h4v1H12zM19 15h2v1H19zM23 15h1v1H23zM4 16h1v1H4zM6 16h2v1H6zM10 16h1v1H10zM12 16h1v1H12zM18,16 h7v1H18zM12 17h2v1H12zM17 17h1v1H17zM22,17 h3v1H22zM4 18h7v1H4zM12 18h3v1H12zM17 18h2v1H17zM21 18h2v1H21zM24,18 h1v1H24zM4 19h1v1H4zM10 19h1v1H10zM13 19h3v1H13zM17 19h2v1H17zM22 19h1v1H22zM4 20h1v1H4zM6 20h3v1H6zM10 20h1v1H10zM12 20h2v1H12zM15 20h1v1H15zM18 20h1v1H18zM20 20h1v1H20zM22 20h2v1H22zM4 21h1v1H4zM6 21h3v1H6zM10 21h1v1H10zM12 21h2v1H12zM16 21h1v1H16zM19 21h2v1H19zM4 22h1v1H4zM6 22h3v1H6zM10 22h1v1H10zM13 22h1v1H13zM15 22h2v1H15zM19 22h3v1H19zM23,22 h2v1H23zM4 23h1v1H4zM10 23h1v1H10zM17 23h1v1H17zM19 23h1v1H19zM21 23h1v1H21zM23,23 h2v1H23zM4 24h7v1H4zM14 24h3v1H14zM18 24h3v1H18zM22 24h2v1H22z"
    fill="#000000"
    shapeRendering="crispEdges"
  />
</svg>
`;

exports[`Pix QRCode QRCode with default arguments 2`] = `
<svg
  height={256}
  viewBox="0 0 29 29"
  width={256}
>
  <path
    d="M0,0 h29v29H0z"
    fill="#FFFFFF"
    shapeRendering="crispEdges"
  />
  <path
    d="M4 4h7v1H4zM13 4h1v1H13zM18,4 h7v1H18zM4 5h1v1H4zM10 5h1v1H10zM14 5h2v1H14zM18 5h1v1H18zM24,5 h1v1H24zM4 6h1v1H4zM6 6h3v1H6zM10 6h1v1H10zM12 6h5v1H12zM18 6h1v1H18zM20 6h3v1H20zM24,6 h1v1H24zM4 7h1v1H4zM6 7h3v1H6zM10 7h1v1H10zM12 7h3v1H12zM16 7h1v1H16zM18 7h1v1H18zM20 7h3v1H20zM24,7 h1v1H24zM4 8h1v1H4zM6 8h3v1H6zM10 8h1v1H10zM13 8h3v1H13zM18 8h1v1H18zM20 8h3v1H20zM24,8 h1v1H24zM4 9h1v1H4zM10 9h1v1H10zM13 9h1v1H13zM15 9h2v1H15zM18 9h1v1H18zM24,9 h1v1H24zM4 10h7v1H4zM12 10h1v1H12zM14 10h1v1H14zM16 10h1v1H16zM18,10 h7v1H18zM7 12h2v1H7zM10 12h2v1H10zM16 12h1v1H16zM21 12h2v1H21zM4 13h2v1H4zM9 13h1v1H9zM11 13h2v1H11zM15 13h2v1H15zM19 13h3v1H19zM23,13 h2v1H23zM4 14h2v1H4zM9 14h4v1H9zM16 14h4v1H16zM21 14h1v1H21zM24,14 h1v1H24zM5 15h1v1H5zM7 15h2v1H7zM12 15h4v1H12zM19 15h2v1H19zM23 15h1v1H23zM4 16h1v1H4zM6 16h2v1H6zM10 16h1v1H10zM12 16h1v1H12zM18,16 h7v1H18zM12 17h2v1H12zM17 17h1v1H17zM22,17 h3v1H22zM4 18h7v1H4zM12 18h3v1H12zM17 18h2v1H17zM21 18h2v1H21zM24,18 h1v1H24zM4 19h1v1H4zM10 19h1v1H10zM13 19h3v1H13zM17 19h2v1H17zM22 19h1v1H22zM4 20h1v1H4zM6 20h3v1H6zM10 20h1v1H10zM12 20h2v1H12zM15 20h1v1H15zM18 20h1v1H18zM20 20h1v1H20zM22 20h2v1H22zM4 21h1v1H4zM6 21h3v1H6zM10 21h1v1H10zM12 21h2v1H12zM16 21h1v1H16zM19 21h2v1H19zM4 22h1v1H4zM6 22h3v1H6zM10 22h1v1H10zM13 22h1v1H13zM15 22h2v1H15zM19 22h3v1H19zM23,22 h2v1H23zM4 23h1v1H4zM10 23h1v1H10zM17 23h1v1H17zM19 23h1v1H19zM21 23h1v1H21zM23,23 h2v1H23zM4 24h7v1H4zM14 24h3v1H14zM18 24h3v1H18zM22 24h2v1H22z"
    fill="#000000"
    shapeRendering="crispEdges"
  />
</svg>
`;
